From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kamillaova <me@kamillaova.dev>
Date: Mon, 12 Feb 2024 13:36:37 +0300
Subject: [PATCH] Call ServerCommandEevent for dispatchCommand


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 4e4906e9765099416b6059c729e133740df64f0a..d416e930ca226deae86078dc67fbc97800b215dd 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1014,6 +1014,13 @@ public final class CraftServer implements Server {
         Preconditions.checkArgument(commandLine != null, "commandLine cannot be null");
         org.spigotmc.AsyncCatcher.catchOp("Command Dispatched Async: " + commandLine); // Spigot // Paper - Include command in error message
 
+        // Cheaf start - call ServerCommandEevent for CraftServer#dispatchCommand
+        var scev = new org.bukkit.event.server.ServerCommandEvent(this.getConsoleSender(), commandLine);
+        this.getPluginManager().callEvent(scev);
+        if (scev.isCancelled()) return true;
+        commandLine = scev.getCommand();
+        // Cheaf end
+
         if (this.commandMap.dispatch(sender, commandLine)) {
             return true;
         }
